pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -121,6 +128,11 @@ func (o *DiffOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	return err
 }
 
+func deleteEmptyAnnotations(obj runtime.Object) {
+	delete(obj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
```

Comment:
You are right. It doesn't work as expected. My intend is just to reuse some code here.

I suspect it may be a bug with `SetAnnotations()`, since it cannot distinguish `SetAnnotations(targetObj, nil)` with `SetAnnotations(targetObj, {})`.
Can you please file an issue for it?